### ISSUE TITLE
Mouse Scroll Stops Working

### ISSUE THREAD

**Issue by arminolat** [zearp/OptiHack#60]
Hello!

I'm experiencing an issue with my Logitech M590 Bluetooth mouse.
This might not be an issue with the OS, but it is something that only happens on my Hackintosh (the mouse works fine on my Windows machine and my work MacBook Pro).

The mouse works perfectly, but after some time the scroll wheel stops working.
Restarting the computer appears to fix the issue, but it happens after a while again.
Middle mouse and all the other buttons works correctly, but scroll stops every few hours.

I am using the device with the Logitech Unifying Wireless Receiver. 

Any help would be much appreciated! 

PS: This is the device I'm using: https://www.logitech.com/en-us/products/mice/m590-silent-wireless-mouse.910-005014.html

**Comment by mgrimace** [comment]
Hello! I'm using the MX Ergo (and spouse uses the MX2 on the same computer) with the Logi unifying receiver and no issues here with scroll wheel on either. As for leads for troubleshooting, are you on the 7020/9020 SFF specifically? I'm wondering if it's the USB map maybe? Do you use Logi Options? I vaguely recall my spouse having some weirdness with the MX when she tried to switch the scrolling direction in the system prefs vs., in Logi Options software. I'm not sure exactly what happened but  we have it set as "natural" in both now.

**Comment by arminolat** [comment]
Hi @mgrimace !

Thanks for replying!

I am using the Optiplex 7020 without hardware modifications.

Logi Options is indeed installed, but I use natural off in both the app and my mouse settings.
Is this software known to cause issues?

**Comment by mgrimace** [comment]
> Logi Options is indeed installed, but I use natural off in both the app and my mouse settings.
> Is this software known to cause issues?

No issues that I'm aware of, I use it myself to remap buttons/gestures/etc., only that I think we ran into problems when scrolling was set at 'natural' in Logi but changed to 'normal' in the system preferences. I don't remember what the problem was TBH though. That was a stretch and I honestly can't think of anything else at the moment if it isn't a hardware thing. Could try another port for the receiver? Maybe on the front vs., the back if there's a range issue (vs., when you use it on your other devices maybe it's closer?). Maybe check in Logi Options after the disconnect and see if the mouse is still registered as 'connected' (e.g., occasionally my mouse works even the it's not showing up as connected in LOGI, in that case some of the custom gestures don't work and I power off/on the mouse). Does a power cycle on the mouse help vs., computer?

**Comment by arminolat** [comment]
@mgrimace these are all good suggestions!

Alas, I tried most of them. Turning the mouse off and on again does not solve anything and it is showing up in Logi options when the issue happens.

**Comment by mgrimace** [comment]
Oh possible solution:

Go to security settings then click privacy tab, then select accessibility. Make sure that the box next to Logitech Options Daemon is checked.

**Comment by arminolat** [comment]
Nice suggestion! But both are enabled already...

![image](https://user-images.githubusercontent.com/24249960/128773236-5c663608-213b-4c21-8946-0ec153f9c6cd.png)

**Comment by zearp** [comment]
Powder cycling the mouse instead of restarting should also work I think. 

To dig down on your problem I would first try to exclude software, removing all Logitech stuff would be the first thing. It should work as normal usb mouse with the unified dongle including the scroll wheel. If it happens without the Logitech stuff installed too we know thats not to blame, if it doesn't happen anymore we know it was the Logitech software.

Does it happen randomly or can you trigger it? Does it happen around the same time after booting up? Like after 10 minutes? Or maybe it always happens when waking up from sleep or when doing a certain thing. If you're actively using the mouse and it happens its not something power management related. It will greatly help in finding the cause if you can trigger it reliably. Did you connect it your MacBook using the dongle too or bluetooth?

Just because it works on your MacBook doesn't mean it's not an OS issue unfortunately. Even if both of your machines have the exact same macOS they are still different in many ways. Different chipsets, drivers, etc etc. Lots of variables. You could try different SMBIOS but I don't think that would matter much. Maybe try older or newer macOS too.

As always the best way to figure it out is to check the logs for any events relating to the mouse. If you can dot down the exact system time when it happens you can find the location in the log files easier and look what happened a few minutes before and after it happens. It's probably the only way to really know what's going on. The Console app, the log/syslog commands in the terminal and grep are your friends.

I did see other people with the same problems, most of the using Logitech and it started around 11.4 -- you didn't specify what you're using and what you've tried so far to rule things out but it might not be a bad idea to open ticket at Logitech, just tell them you have a Mac mini or whatever SMBIOS you're using, they surly know what the common causes of it are and/or can help getting more/better logs. Maybe their drivers can have some debug like logging if they're not the cause.

I don't think this issue is related to the EFI/config as more people have this exact same issue. I also had my mouse wheel stop but only after wake-ups and very rarely and I'm using bluetooth (which also goes over usb). Never happened on Catalina using the same machine, same EFI and same mouse. But it does happen every now and then in Big Sur. Haven't tried Monterey on that machine yet. But the same mouse works fine on other machines. macOS can be very annoying in this way.

The previous version of the Microsoft surface mouse did some very weird things on my MacBook like disconnecting and reconnecting every 10-15 minutes or stopped working until I tuned the mouse off and on but it worked perfectly fine on my hacks. Then the new version of that mouse with a metal scroll wheel instead of a plastic one works fine on both. Unrelated to this but shows how tricky it can be. Sometimes there really is no sense or logic in it and it just doesn't work lol.

tl;dr to get to the bottom of it you will need to do a lot of testing and wading trough log files but before putting in too much time I would take advantage of the Logitech support first. Open ticket so you can get referred more easily to people who know about debugging drivers. Community forum are useless for anything more technical imho and calling will not you get anyone beyond someone in a call centre working from a script.

(I'm closing this issue but feel free to update us, closing it doesn't disable new replies. I just want to reserve open issues for issues caused by the EFI/config.)
